### PR TITLE
Tech debt upgrade

### LINKED ISSUE #153: Upgrade CAP to Java 11 and double-check Maintenance
UI
1. upgrade dependencies
2. update auth service
3. update/verify build and ci

Service
1. upgrade weaver
2. update auth service
3. update/verify build and ci

Update POM repositories:

```
    <repository>
      <id>tamu-releases</id>
      <url>https://artifacts.library.tamu.edu/repository/maven-releases/</url>
    </repository>

    <repository>
      <id>tamu-snapshots</id>
      <url>https://artifacts.library.tamu.edu/repository/maven-snapshots/</url>
    </repository>
```

Update Weaver Webservice Parent:

```
  <parent>
    <groupId>edu.tamu.weaver</groupId>
    <artifactId>webservice-parent</artifactId>
    <version>2.0.2</version>
  </parent>
```

Update Weaver UI Core:

`"@wvr/core": "2.0.6"`

### PR DESCRIPTION
**Pull request by rladdusaw**
Resolves #153

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -98,13 +100,13 @@
     <dependency>
       <groupId>xml-apis</groupId>
       <artifactId>xml-apis</artifactId>
-      <version>1.4.01</version><!--$NO-MVN-MAN-VER$ -->
```
Comment (kaladay) [inline pom.xml:101]: The original version (1.4.01) is actually correct and more recent than 2.0.2.
We need to add a comment to document this confusion.

File: pom.xml [hunk 2]
```diff
@@ -98,13 +100,13 @@
...
+      <version>2.0.2</version>      <!--$NO-MVN-MAN-VER$ -->
     </dependency>
 
     <dependency>
       <groupId>com.github.jsonld-java.jsonld-java</groupId>
       <artifactId>jsonld-java</artifactId>
-      <version>0.12.5</version>
+      <version>0.13.4</version>
```
Comment (kaladay) [inline pom.xml:109]: An exclusion for jackson-databind is necessary due to a vulnerability in it:
- [CVE-2020-36518](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-36518)

We should explicitly include a version like 2.13.3.

File: pom.xml [hunk 3]
```diff
@@ -162,8 +160,8 @@
       <scope>test</scope>
       <exclusions>
         <exclusion>
-          <groupId>org.springframework.boot</groupId>
-          <artifactId>spring-boot-starter-logging</artifactId>
+          <groupId>org.junit.vintage</groupId>
```
Comment (kaladay) [inline pom.xml:163]: Vintage no longer needs to be excluded.
The latest spring boot starter test does not include it.
Please remove the exclusion.

File: src/main/java/edu/tamu/cap/config/AppH2ConsoleConfig.java [hunk 4]
```diff
@@ -11,8 +11,8 @@
 public class AppH2ConsoleConfig {
 
     @Bean
-    public ServletRegistrationBean h2servletRegistration() {
-        ServletRegistrationBean registrationBean = new ServletRegistrationBean(new WebServlet());
+    public ServletRegistrationBean<WebServlet> h2servletRegistration() {
+        ServletRegistrationBean<WebServlet> registrationBean = new ServletRegistrationBean<WebServlet>(new WebServlet());
```
Comment (kaladay) [inline src/main/java/edu/tamu/cap/config/AppH2ConsoleConfig.java:15]: Should be able to change `<WebServlet>` to `<>` on the right side of the equal.

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+# use this as the security key for encryption
+security:
+  user:
+    password: 81d56928-891a-41b2-8dd2-132593e6d057
+
+spring:
+  main:
+    allow-bean-definition-overriding: true
+    allow-circular-references: true
```
Comment (kaladay) [inline src/main/resources/application.yml:13]: There needs to be an issue created for the `    allow-circular-references: true`.

File: pom.xml [hunk 6]
```diff
@@ -489,82 +270,47 @@
...
-            </configuration>
           </execution>
         </executions>
       </plugin>
-      <plugin> 
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-resources-plugin</artifactId>
+        <version>3.2.0</version>
```
Comment (kaladay) [inline pom.xml:297]: This version is already in the managed version and should be removed.

File: src/test/java/edu/tamu/cap/auth/service/AppUserCredentialsServiceTest.java [hunk 7]
```diff
@@ -51,7 +51,7 @@ public final class AppUserCredentialsServiceTest {
 
     @BeforeEach
     public void setup() throws JsonParseException, JsonMappingException, IOException {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (kaladay) [inline src/test/java/edu/tamu/cap/auth/service/AppUserCredentialsServiceTest.java:54]: We shouldn't be using either of these:
```
        MockitoAnnotations.initMocks(this);
        MockitoAnnotations.openMocks(this);
```

instead, prefix the class with this annotation:
```
@ExtendWith(MockitoExtension.class)
```

Also, the `@SpringBootTest` annotation already includes `@ExtendWith(SpringExtension.class)`, so please remove this from line ~31.

File: src/test/java/edu/tamu/cap/controller/SchemaControllerTest.java [hunk 8]
```diff
@@ -42,7 +45,7 @@
 
 
 @SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
```
Comment (kaladay) [inline src/test/java/edu/tamu/cap/controller/SchemaControllerTest.java:48]: `@SpringBootTest` already includes this, please remove `ExtendWith(SpringExtension.class)`.

File: src/test/java/edu/tamu/cap/controller/UserControllerTest.java [hunk 9]
```diff
@@ -20,17 +22,15 @@
...
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-
 import edu.tamu.cap.model.User;
 import edu.tamu.cap.model.repo.UserRepo;
 import edu.tamu.cap.utility.MockUserUtility;
 
 @SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
```
Comment (kaladay) [inline src/test/java/edu/tamu/cap/controller/UserControllerTest.java:33]: `@SpringBootTest` already includes this, please remove `ExtendWith(SpringExtension.class)`.

File: src/test/java/edu/tamu/cap/controller/VerifyRepositoryViewSettingsControllerTest.java [hunk 10]
```diff
@@ -29,19 +33,17 @@
...
-
 import edu.tamu.cap.model.RepositoryView;
 import edu.tamu.cap.model.repo.RepositoryViewRepo;
 import edu.tamu.cap.service.RepositoryViewType;
 import edu.tamu.cap.service.repositoryview.FedoraService;
 import edu.tamu.cap.utility.ConstraintDescriptionsHelper;
 
 @SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
```
Comment (kaladay) [inline src/test/java/edu/tamu/cap/controller/VerifyRepositoryViewSettingsControllerTest.java:46]: `@SpringBootTest` already includes this, please remove `ExtendWith(SpringExtension.class)`.

File: src/test/java/edu/tamu/cap/service/FedoraServiceTest.java [hunk 11]
```diff
@@ -32,6 +33,7 @@
 import edu.tamu.cap.model.response.Version;
 import edu.tamu.cap.service.repositoryview.FedoraService;
 
+@ActiveProfiles("test")
 @ExtendWith(SpringExtension.class)
```
Comment (kaladay) [inline src/test/java/edu/tamu/cap/service/FedoraServiceTest.java:37]: `@SpringBootTest` already includes this, please remove `ExtendWith(SpringExtension.class)`.

File: src/test/java/edu/tamu/cap/service/FedoraServiceTest.java [hunk 12]
```diff
@@ -79,7 +81,7 @@ public void setup() throws IOException {
 
         when(request.getInputStream()).thenReturn(null);
 
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (kaladay) [inline src/test/java/edu/tamu/cap/service/FedoraServiceTest.java:84]: Remove this and instead add:
```
ExtendWith(MockitoExtension.class)
```

File: src/test/resources/application.yml [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+  level:
+    edu:
+      tamu: INFO
+    org:
+      hibernate: INFO
+      springframework: INFO
+    ro:
+      isdc:
+        wro: INFO
+  file.name: logs/cap.log
```
Comment (kaladay) [inline src/test/resources/application.yml:67]: I started a practice of adding `-test` to the test log file.
I request changing this to:
```
file.name: logs/cap-test.log
```

**Review by kaladay — CHANGES_REQUESTED** [review]
In the pom.xml, code like:
```
              <!-- NPM Install -->
              <execution>
                <id>exec-npm-install</id>
                <phase>generate-sources</phase>
...
```

According to eclipse:
```
Plugin execution not covered by lifecycle configuration: org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (execution: exec-npm-install, phase: generate-sources)
```
For lines: 529, 547, 562, 578.
I do not know if this is true/false so I figured I would mention it.

I get the following:
```
ServletRegistrationBean is a raw type. References to generic type ServletRegistrationBean<T> should be parameterized
```
for:
```
    @Bean
    public ServletRegistrationBean h2servletRegistration() {
        ServletRegistrationBean registrationBean = new ServletRegistrationBean(new WebServlet());
```

I replaced this with:
```
    @Bean
    public ServletRegistrationBean<WebServlet> h2servletRegistration() {
        ServletRegistrationBean<WebServlet> registrationBean = new ServletRegistrationBean<WebServlet>(new WebServlet());
```
If this is a correct thing to do, then I suggest such a change.


The 
```
private final Logger logger = LoggerFactory.getLogger(this.getClass());
```
in `MapService.java` is not used.
I suggest considering removing it.

File: Dockerfile [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+COPY ./src ./src
+COPY ./package.json ./package.json
+COPY ./Gruntfile.js ./Gruntfile.js
+COPY ./.jshintrc ./.jshintrc
+
+# Assign file permissions.
+RUN chown -R ${USER_ID}:${USER_ID} ${SOURCE_DIR}
+
+# Install grunt.
+RUN npm install -g grunt-cli
```
[the remark this answers is not in the document]
Comment (rladdusaw) [reply Dockerfile:41]: It is necessary, you can't install a global package without root.

**Comment by kaladay** [comment]
I also suggest adding a logback file to further simplify the output when running tests.
Here is an example I copied from iriif:
`src/test/resources/logback-test.xml`:
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <include resource="org/springframework/boot/logging/logback/base.xml" />
  <logger name="com.zaxxer" level="ERROR"/>
  <logger name="org.hibernate" level="OFF"/>
  <logger name="org.springframework" level="ERROR"/>
  <logger name="edu.tamu.iiif" level="WARN"/>
  <property name="root.level.console" value="OFF"/>
</configuration>

```

File: pom.xml [hunk 15]
```diff
@@ -1,12 +1,12 @@
...
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
   <groupId>edu.tamu</groupId>
   <artifactId>cap</artifactId>
   <version>0.6.3-SNAPSHOT</version>
-  <packaging>war</packaging>
+  <packaging>${packaging}</packaging>
```
Comment (kaladay) [inline pom.xml:9]: I request doing the following to be consistent.
Change packaging in this manner:
```
  <properties>
    <maven.packaging>war</maven.packaging>
  </properties>

  <packaging>${maven.packaging}</packaging>

  <profiles>
    <profile>
      <id>war</id>
      <properties>
        <maven.packaging>war</maven.packaging>
      </properties>
    </profile>
    <profile>
      <id>jar</id>
      <properties>
        <maven.packaging>jar</maven.packaging>
      </properties>
    </profile>
  </profiles>
  ```

Comment (kaladay) [reply pom.xml:9]: You can also do this:
```
  <properties>
    <maven.packaging>${packaging}</maven.packaging>
  </properties>
  ```
  Preserving the behavior of using `${packagin}` by default.

File: pom.xml [hunk 16]
```diff
@@ -893,37 +745,46 @@
...
 
   </profiles>
 
+  <reporting>
+    <outputDirectory>${project.build.directory}/site</outputDirectory>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-project-info-reports-plugin</artifactId>
+        <version>3.2.2</version>
```
Comment (kaladay) [inline pom.xml:760]: See if you can use version `3.0.0` as it doesn't bring in dependencies with CVEs at this time.

File: pom.xml [hunk 17]
```diff
@@ -577,36 +323,165 @@
...
+              <doctype>book</doctype>
+              <sourceDocumentName>index.adoc</sourceDocumentName>
+            </configuration>
+          </execution>
+        </executions>
+        <dependencies>
+          <dependency>
+            <groupId>org.springframework.restdocs</groupId>
+            <artifactId>spring-restdocs-asciidoctor</artifactId>
+            <version>2.0.3.RELEASE</version>
```
Comment (kaladay) [inline pom.xml:477]: I think we should use version `<version>2.0.6.RELEASE</version>`.
None of its dependencies currently have a CVE as `2.0.3.RELEASE` does.

File: src/main/resources/application.yml [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+    url: jdbc:h2:mem:AZ;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE
+    driverClassName: org.h2.Driver
+    # PostgreSQL
+    # url: jdbc:postgresql://localhost:5432/cap
+    # driverClassName: org.postgresql.Driver
+    # MySQL
+    # url: jdbc:mysql://localhost/cap
+    # driverClassName: com.mysql.jdbc.Driver
+    jpa:
+      database-platform: org.hibernate.dialect.H2Dialect
```
Comment (kaladay) [inline src/main/resources/application.yml:36]: This is incorrect.

The `spring.datasource.jpa.dabase-platform` needs to be changed to `spring.jpa.database-platform`.

There is a `jpa` around line 46 below that this should be moved to.

see: [spring.jpa.database-platform](https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html#application-properties.data.spring.jpa.database-platform)

File: src/main/resources/application.yml (hunk 5 above)
Comment (kaladay) [reply src/main/resources/application.yml:13]: I went ahead and created the issue here: #178 .

File: pom.xml (hunk 2 above)
Comment (kaladay) [reply pom.xml:109]: This can be solved by switching to the new Spring-boot 2.7.0 (or greater).

File: src/test/resources/application.yml [hunk 19]
```diff
@@ -96,7 +96,7 @@ app:
     replyTo: dev@library.tamu.edu
   # edu.tamu.weaver.reporting.controller.ReportingController
   reporting:
-    address: dev@tdl.or
+    address: dev@tdl.org
```
Comment (wwelling) [inline src/test/resources/application.yml:99]: Let's update this and above to `helpdesk@library.tamu.edu`. It is a overlooked copy and paste from Vireo.

File: pom.xml [hunk 20]
```diff
@@ -23,48 +21,56 @@
...
-      <version>2.1.1-RC9</version>
+      <version>${project.parent.version}</version>
     </dependency>
 
     <dependency>
       <groupId>edu.tamu.weaver</groupId>
       <artifactId>wro</artifactId>
-      <version>2.1.1-RC9</version>
+      <version>${project.parent.version}</version>
+      <exclusions>
```
Comment (wwelling) [inline pom.xml:56]: This exclusion workaround is corrected in latest release of weaver and can be removed.

**Review by kaladay — APPROVED** [review]

**Pull request merged**